sdks/unity/AgonesSdk.cs

```diff
@@ -38,36 +38,47 @@ public class AgonesSdk : MonoBehaviour
         [Range(0.01f, 5)] public float healthIntervalSecond = 5.0f;
 
         /// <summary>
-        /// Whether the server sends a health ping to the Agones sidecar.
+        /// Whether to start sending health ping to the Agones sidecar.
         /// </summary>
-        public bool healthEnabled = true;
+        [SerializeField] private bool startHealthChecks = true;

```

Comment:
Is there a good reason to change this name?

This would be a breaking change for the SDK, so we would like to avoid that where possible.

Reply:
Also, changing the encapsulation would also be a breaking change. Is there a good reason to do that as well?

Reply:
> Is there a good reason to change this name?
The name itself is misleading, for example, the user may set it to true externally an expect health checks to start, which won't work because the variable is **only** checked in the Start method, in case if it was set to true then health checks start.

> Also, changing the encapsulation would also be a breaking change. Is there a good reason to do that as well?

This variable was made public so unity will be able to serialize and show it int the Inspector it basically has no external use case, see code examples below.

```
void Start()
{
    sdk.healthEnabled = true; //There is a 50 % chance that the health checks will start due to Unity random scheduling. 
}

void Update()
{
    if(someCondition)
        sdk.healthEnabled = true; //This won't work. 

    if(someOtherCondition)
        sdk.healthEnabled = false; //This doesn't make sense because we can't renew the health checks, instead we can use sdk.Shutdown() 
}
```

Reply:
To be honest, I'm not sure why you would be trying to dynamically decide if health checking is enable or not - you would enable health checking if you have it turned on on Agones. That's really it.

Health checking is not a tool for managing game server lifecycles.